src/main/resources/com/google/api/codegen/ruby/Rakefile.snip

```diff
@@ -1,5 +1,76 @@
+@extends "ruby/common.snip"
+
 @snippet generate(metadata)
-  @# -*- ruby -*-
-  require 'bundler/gem_tasks'
+  {@header(metadata.fileHeader)}
+
+  {@unitTestsTask(metadata)}
+  @if smokeTestsTask(metadata)
+
+    {@smokeTestsTask(metadata)}
+  @else
+  @end
+@end
+
+@private header(fileHeader)
+  {@toComments(fileHeader.copyrightLines)}
+  #
+  {@toComments(fileHeader.licenseLines)}
+
+  require "bundler/setup"
+  require "bundler/gem_tasks"
+
+  require "rubocop/rake_task"
+  RuboCop::RakeTask.new
+@end
+
+@private unitTestsTask(metadata)
+  desc "Run tests."
+  task :test do
+    $LOAD_PATH.unshift "lib", "test"
+    Dir.glob("test/**/*_test.rb").each { |file| require_relative file }
+  end
+
+  namespace :test do
+    desc "Runs tests with coverage."
+    task :coverage do
+      require "simplecov"
+      SimpleCov.start do
+        command_name "{@metadata.identifier}"
+        track_files "lib/**/*.rb"
+        add_filter "test/"
+      end
+
+      Rake::Task["test"].invoke
+    end
+  end
+@end
+
+@private smokeTestsTask(metadata)
+  @if metadata.hasSmokeTests
+    desc "Runs the smoke tests."
+    task :smoke_test, :project do |t, args|
+      ENV["SMOKE_TEST_PROJECT"] = :project || ENV["SMOKE_TEST_PROJECT"]
```

Comment:
Ah this is correct. This will also mean that the smoketests will need to be updated by creating credentials with the project ID and and the keyfil. I don't seem the project id being used credentials wise here: [src/test/java/com/google/api/codegen/testdata/ruby_smoke_test_library.baseline#L27](https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/ruby_smoke_test_library.baseline#L27)

Creation of the credentials will look like so: https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/ea2be47beb32615b2bf69f8a846a684f86c8328c/google-cloud-vision/lib/google/cloud/vision/credentials.rb